waterbox/libcxx/configure-for-waterbox-phase--

```diff
@@ -2,25 +2,34 @@
 set -e
 if [ -z "$SYSROOT" ]; then export SYSROOT="$(realpath "$(dirname "$0")/../sysroot")"; fi
 if [ -z "$LLVMDIR" ]; then export LLVMDIR="$(realpath "$(dirname "$0")/../llvm-project")"; fi
+if [ -f "$SYSROOT/bin/musl-gcc" ]; then export CC="$SYSROOT/bin/musl-gcc"; fi
+if [ -f "$SYSROOT/bin/musl-clang" ]; then export CC="$SYSROOT/bin/musl-clang"; fi
 
 rm -rf build-
 mkdir build-
 cd build-
-export CFLAGS="-mcmodel=large -mstack-protector-guard=global -no-pie -fno-pic -fno-pie"
-export CXXFLAGS="-mcmodel=large -mstack-protector-guard=global -no-pie -fno-pic -fno-pie"
+export ASMFLAGS="-w -mcmodel=large -mstack-protector-guard=global -fno-pic -fno-pie -fcf-protection=none"
+export CFLAGS="-w -mcmodel=large -mstack-protector-guard=global -fno-pic -fno-pie -fcf-protection=none"
+export CXXFLAGS="-w -mcmodel=large -mstack-protector-guard=global -fno-pic -fno-pie -fcf-protection=none -fno-use-cxa-atexit"
+export LDFLAGS="-no-pie"
+
 cmake \
-	-DCMAKE_C_COMPILER="$SYSROOT/bin/musl-gcc" \
-	-DCMAKE_CXX_COMPILER="$SYSROOT/bin/musl-gcc" \
+	-DCMAKE_ASM_COMPILER="$CC" \
```

Comment:
cmake warns that `CMAKE_ASM_COMPILER` is unused here (and in the 1 and 2 scripts as well), so it's probably best to remove the `ASMFLAGS` and `CMAKE_ASM_COMPILER` specifications.

Reply:
I'm assuming you meant to review another file since compiler_rt does have assembly bits (and thus cmake gives no warnings here), unless something weird with your setup is causing it to not use assembly bits here? Only 1 and 2 have asm compiler unused.

Reply:
I didn't actually run through the entire libcxx setup and was instead just going by memory, my bad 😅